### ISSUE TITLE
1/0 Not Converting to Boolean on Model.Create()

### ISSUE THREAD

**Issue by sailsbot** [balderdashy/sails#4007]
_From @jvanalst on September 7, 2016 16:34_

**Waterline version**: 0.11.2
**Node version**: v4.5.0
**NPM version**: 2.15.9
**Operating system**: Linux-like (Ubuntu, Mac OS X)

<hr/>

This is as much a question as anything else.

Steps to replicate:

``` bash
  sails new boolean-test
  sails generate api boolean
```

/api/models/Boolean.js :

``` js
//boolean.js
module.exports = {

  attributes: {
    test: {
      type: 'boolean',
      required: true
    }
  }

};
```

When I call a `model.create()` and POST a "1" or "0" representing my boolean it blows up with a WL validation error. It doesn't do the type conversion automatically. 

E.g. `http://localhost:1337/boolean/create?test=1` 
returns ``undefined` should be a boolean (instead of "1", which is a string)`

But when I do a `model.find()` with a "1"/"0" it converts to a boolean just fine and finds what I'm looking for.  E.g. `http://localhost:1337/boolean/?test=1` returns all the boolean objects where test is true.

Is this inconsistency intended? Or is it a bug?

_Copied from original issue: balderdashy/waterline#1391_

**Comment by sailsbot** [comment]
Hi @jvanalst!  It looks like you may have removed some required elements from the initial comment template, without which I can't verify that this post meets our [contribution guidelines](http://bit.ly/sails-issue-guide). To re-open this issue, please copy the template from [here](https://raw.githubusercontent.com/balderdashy/waterline/master/.github/ISSUE_TEMPLATE), paste it at the beginning of your initial comment, and follow the instructions in the text. Then post a new comment (e.g. "ok, fixed!") so that I know to go back and check.

Sorry to be a hassle, but following these instructions ensures that we can help you in the best way possible and keep the Sails project running smoothly.

_If you feel this message is in error, or you want to debate the merits of my existence (sniffle), please contact inquiries@treeline.io._

**Comment by sailsbot** [comment]
_From @jvanalst on September 7, 2016 16:37_

@sailsbot fixed?

**Comment by sailsbot** [comment]
Sorry to be a hassle, but it looks like your issue is still missing some required info.  Please double-check your initial comment and try again.

_If you feel this message is in error, or you want to debate the merits of my existence (sniffle), please contact inquiries@treeline.io._

**Comment by sailsbot** [comment]
_From @jvanalst on September 7, 2016 16:37_

@sailsbot How about now?

**Comment by sailsbot** [comment]
@jvanalst,@sailsbot: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).

If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by sailsbot** [comment]
_From @jvanalst on October 11, 2016 15:22_

The inconsistency still hasn't been addressed. I have no further information.

**Comment by sailsbot** [comment]
@jvanalst,@sailsbot: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).

If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by sailsbot** [comment]
_From @jvanalst on November 14, 2016 15:50_

Still no clarification.

**Comment by sailsbot** [comment]
@jvanalst,@sailsbot: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by sailsbot** [comment]
_From @jvanalst on December 15, 2016 23:23_

Still no clarification.

**Comment by sailsbot** [comment]
@jvanalst,@sailsbot: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by sailsbot** [comment]
_From @jvanalst on January 16, 2017 16:32_

Still no clarification.

**Comment by sailsbot** [comment]
@jvanalst,@sailsbot: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by sailsbot** [comment]
_From @jvanalst on February 16, 2017 23:28_

bot fail

**Comment by sailsbot** [comment]
@jvanalst It sounds like you've isolated an inconsistency inherent in the way that Sails v0.12 blueprint actions do their parsing.  Would you mind trying this out with Sails v1 for clarification and posting your results back here?  

I hope I haven't frustrated you too much with all of the warnings about closing the issue.  Thanks again for keeping this open!

**Comment by sailsbot** [comment]
Thanks for posting, @sailsbot.  I'm a repo bot-- nice to meet you!

The issue queue in this repo is for verified bugs with documented features.  Unfortunately, we can't leave some other types of issues marked as "open".  This includes bug reports about undocumented features or unofficial plugins, as well as feature requests, questions, and commentary about the core framework. Please review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) for details.

If you're here looking for help and can't find it, visit [StackOverflow](http://stackoverflow.com), our [Google Group](https://groups.google.com/forum/#!forum/sailsjs) or our [chat room](https://gitter.im/balderdashy/sails?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge).  But please do not let this disrupt future conversation in this issue! I am only marking it as "closed" to keep organized.

Thanks so much for your help!

> If I am mistaken, and this issue is about a _critical bug with a documented feature_, please accept my sincerest apologies-- I am but a humble bot working to make GitHub a better place.  If you open a new issue, a member of the core team will take a look ASAP.
